gdsfactory/components/coupler_adiabatic.py

```diff
@@ -50,44 +75,272 @@ def coupler_adiabatic(
+    ]
+
+    control_points_input_bottom = [
+        (0, -input_wg_sep),
+        (length1 / 2.0, -input_wg_sep),
+        (length1 / 2.0, -input_wg_sep / 2.0 - wg_sep / 2.0),
+        (length1, -input_wg_sep / 2.0 - wg_sep / 2.0),
+    ]
+
+    control_points_output_top = [
+        (length1 + length2, -input_wg_sep / 2.0 + wg_sep / 2.0),
+        (
+            length1 + length2 + length3 / 2.0,
+            -input_wg_sep / 2.0 + wg_sep / 2.0,
+        ),
+        (
+            length1 + length2 + length3 / 2.0,
+            -input_wg_sep / 2.0 + output_wg_sep / 2.0,
+        ),
+        (
+            length1 + length2 + length3,
+            -input_wg_sep / 2.0 + output_wg_sep / 2.0,
+        ),
+    ]
+
+    control_points_output_bottom = [
+        (length1 + length2, -input_wg_sep / 2.0 - wg_sep / 2.0),
+        (
+            length1 + length2 + length3 / 2.0,
+            -input_wg_sep / 2.0 - wg_sep / 2.0,
+        ),
+        (
+            length1 + length2 + length3 / 2.0,
+            -input_wg_sep / 2.0 - output_wg_sep / 2.0,
+        ),
+        (
+            length1 + length2 + length3,
+            -input_wg_sep / 2.0 - output_wg_sep / 2.0,
+        ),
+    ]    
+
+    # Bend specific bezier functions
+    def _bezier_inp_top(t):
+        return _bezier_curve(t, control_points_input_top)
+
+    def _bezier_inp_bottom(t):
+        return _bezier_curve(t, control_points_input_bottom)
+
+    def _bezier_output_bottom(t):
+        return _bezier_curve(t, control_points_output_bottom)
+
+    def _bezier_output_top(t):
+        return _bezier_curve(t, control_points_output_top)
+
+    c = Component()
+
+    x = gf.get_cross_section(cross_section, **kwargs)
+
+    wg_width = x.width
+
+    c.add(_generate_S_bend(wg_width=wg_width, bezier_function=_bezier_inp_top, final_width=wg_width + dw / 2.0, layer=gf.get_layer(x.layer)[0], datatype=gf.get_layer(x.layer)[1]))
+
+    # Bottom input S-bend
+    c.add(_generate_S_bend(wg_width=wg_width, bezier_function=_bezier_inp_bottom, final_width=wg_width - dw / 2.0, layer=gf.get_layer(x.layer)[0], datatype=gf.get_layer(x.layer)[1]))
+
+    # Top output S-bend
+    c.add(_generate_S_bend(wg_width=wg_width, bezier_function=_bezier_output_top, final_width=wg_width, layer=gf.get_layer(x.layer)[0], datatype=gf.get_layer(x.layer)[1]))
+
+    # Bottom output S-bend
+    c.add(_generate_S_bend(wg_width=wg_width, bezier_function=_bezier_output_bottom, final_width=wg_width, layer=gf.get_layer(x.layer)[0], datatype=gf.get_layer(x.layer)[1]))
+
+    # Get cladding layers
+    try:
+        if kwargs["cladding_layers"] is None:
+            kwargs["cladding_layers"] = (111, 0)
+        else:
+            layers = []
+            for layer in kwargs["cladding_layers"]:
+                layers.append(gf.get_layer(layer))
+            kwargs["cladding_layers"] = layers
+    except KeyError:
+        kwargs["cladding_layers"] = (111, 0)
+
+    # Instantiate S-bend cladding objects
+    try:
```

Comment:
the claddings will be added in the child cells when you extrude a Path

you can take a look at `gf.components.bend_s`